src/aws-cpp-sdk-core/source/http/crt/CRTHttpClient.cpp

```diff
@@ -246,13 +246,19 @@ namespace Aws
             }
 
             //TODO: handle the read rate limiter here, once back pressure is setup.
+            assert(response);
             for (const auto& hashIterator : request->GetResponseValidationHashes())
```

Comment:
An improvement here would be to delay hash initialization until it's first used.